library/src/main/java/com/schibsted/spain/barista/internal/viewaction/ChildActions.kt

```diff
@@ -0,0 +1,56 @@
+package com.schibsted.spain.barista.internal.viewaction
+
+import android.support.annotation.IdRes
+import android.support.test.espresso.PerformException
+import android.support.test.espresso.UiController
+import android.support.test.espresso.ViewAction
+import android.support.test.espresso.matcher.ViewMatchers.hasDescendant
+import android.support.test.espresso.matcher.ViewMatchers.isDisplayed
+import android.support.test.espresso.matcher.ViewMatchers.withId
+import android.support.test.espresso.util.HumanReadables
+import android.view.View
+import com.schibsted.spain.barista.internal.failurehandler.description
+import com.schibsted.spain.barista.internal.util.ViewTreeAnalyzer
+import org.hamcrest.Matcher
+import org.hamcrest.Matchers.allOf
+
+object ChildActions {
+
+  @JvmStatic
+  fun performOnChildWithId(@IdRes childId: Int, viewAction: ViewAction): ViewAction {
```

Comment:
The `perform` word here could cause some confusion. Because this function doesn't really perform anything, it just returns the ViewAction that will be passed to Espresso's `perform()` method.

What do you think about naming it simply `onChildWithId(...)`? That way the common usage would look like `onView(...).perform(onChildWithId(R.id.child, click())`

Reply:
Makes sense, I'll change it 👍